core/tests/protractor_desktop/SkillEditorPage.js

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for the skill editor page.
+ */
+
+var forms = require('../protractor_utils/forms.js');
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var waitFor = require('../protractor_utils/waitFor.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var TopicsAndSkillsDashboardPage =
+  require('../protractor_utils/TopicsAndSkillsDashboardPage.js');
+var SkillEditorPage =
+  require('../protractor_utils/SkillEditorPage.js');
+
+describe('Skill Editor functionality', function() {
+  var topicsAndSkillsDashboardPage = null;
+  var skillEditorPage = null;
+
+  beforeAll(function() {
+    users.createAdmin('test@example.com',
+      'skillEditor');
+    topicsAndSkillsDashboardPage =
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage();
+    skillEditorPage =
+      new SkillEditorPage.SkillEditorPage();
+  });
+
+  beforeEach(function() {
+    users.login('test@example.com');
+
+    topicsAndSkillsDashboardPage.get();
+    topicsAndSkillsDashboardPage.createSkillWithDescription('Test Skill');
```

Comment:
Is this supposed to be in `beforeEach()` or `beforeAll()`? Since, I think all the tests can be done on just one skill, or is that not the case?

Reply:
Yes, all tests _can_ be done on just one skill. Do you prefer the `beforeAll` version?

Reply:
Yeah, since otherwise the tests would become unnecessarily slow.